test/rules/promise-function-async/test.ts.lint

```diff
@@ -0,0 +1,65 @@
+declare class Promise<T>{}
+
+const nonAsyncPromiseFunctionExpressionA = function(p: Promise<void>) { return p; };
```

Comment:
for every test that causes a lint failure, you should have a matching test that passes